File: kgparserSpringBoot/src/main/java/ru/klavogonki/kgparser/zip/ZipUtils.java
package ru.klavogonki.kgparser.zip;

import lombok.extern.log4j.Log4j2;
import net.lingala.zip4j.ZipFile;
import net.lingala.zip4j.exception.ZipException;

@Log4j2
public final class ZipUtils {

    private ZipUtils() {
    }

    public static void zipFile(String filePath, String zipFilePath) {
        try {
            new ZipFile(zipFilePath).addFile(filePath);
            logger.debug("Successfully zipped file {} to zip file {}", filePath, zipFilePath);
        }
        catch (ZipException e) {
            logger.error(String.format("Error on zipping file %s to zip file %s", filePath, zipFilePath), e );

            throw new RuntimeException(e);
        }
    }
}

File: kgstatsSrv/src/main/kotlin/ru/klavogonki/statistics/zip/ZipUtils.kt
package ru.klavogonki.statistics.zip

import net.lingala.zip4j.ZipFile
import net.lingala.zip4j.exception.ZipException
import org.apache.logging.log4j.kotlin.Logging

object ZipUtils : Logging {

    @JvmStatic
    fun zipFile(filePath: String, zipFilePath: String) {
        ZipFile(zipFilePath).use {
            try {
                it.addFile(filePath)

                logger.debug("Successfully zipped file \"$filePath\" to zip file \"$zipFilePath\".")
            } catch (e: ZipException) {
                logger.error("Error on zipping file \"$filePath\" to zip file \"$zipFilePath\"", e)

                throw e
            }
        }
    }
}